subsys/bluetooth/mesh/dfd_srv.c

```diff
@@ -925,12 +925,34 @@ const struct bt_mesh_blob_srv_cb _bt_mesh_dfd_srv_blob_cb = {
 
 static int dfd_srv_init(const struct bt_mesh_model *mod)
 {
+	int err;
 	struct bt_mesh_dfd_srv *srv = mod->rt->user_data;
 
 	srv->mod = mod;
 
-	if (IS_ENABLED(CONFIG_BT_MESH_MODEL_EXTENSIONS)) {
-		bt_mesh_model_extend(mod, srv->upload.blob.mod);
+	const struct bt_mesh_model *blob_srv =
+		bt_mesh_model_find(bt_mesh_model_elem(mod), BT_MESH_MODEL_ID_BLOB_SRV);
+
+	if (blob_srv == NULL) {
+		LOG_ERR("Missing BLOB Srv on the same element.");
```

Comment:
"on the same element" is unnecessary IMO. The code is quite self explanatory for anyone to understand what is wrong. With today's AI tools, with the context of "Missing BLOB Srv", any developer can understand what is missing. I will really prefer to keep things concise.

Reply:
OK; IMO the log messages are OK as it is.

> "on the same element" is unnecessary IMO.

Necessary IMO.

> The code is quite self explanatory for anyone to understand what is wrong.

In this regard removing log messages seems quite reasonable, I agree.

> With today's AI tools, with the context of "Missing BLOB Srv", any developer can understand what is missing. I will 
really prefer to keep things concise.

I don't understand this, can we decide on one thing, I feel like the discussion is beyond the scope, we can "discuss how a precise log message should be" somewhere else.

Reply:
https://github.com/nrfconnect/sdk-nrf/blob/046312b6f9c008541e996cf603bf518a02e6503a/subsys/bluetooth/mesh/light_hsl_srv.c#L597

https://github.com/nrfconnect/sdk-nrf/blob/046312b6f9c008541e996cf603bf518a02e6503a/subsys/bluetooth/mesh/light_xyl_srv.c#L659

🙃

Reply:
I reallyy dislike all of those now after seeing how footprint balloons.

Reply:
Let's to not abuse this PR for preferences. I like long unique comments those describe what happens. Kind of disagree to shortened forms since they require additional investigation during debugging since this is not clear which one has been triggered.

Reply:
> I reallyy dislike all of those now after seeing how footprint balloons.

This is an open source project. The message should be clear even for developers who didn't spend years for Mesh.